dockertest/subtest.py

```diff
@@ -19,16 +19,45 @@
 from ConfigParser import Error
 from autotest.client.shared.error import TestError, TestNAError
 from autotest.client.shared.version import get_version
-from autotest.client import test
+from autotest.client import test, utils
 import version
 import config
 import subtestbase
+from docker_daemon import which_docker
 from xceptions import DockerTestFail
 from xceptions import DockerTestNAError
 from xceptions import DockerTestError
 from xceptions import DockerSubSubtestNAError
 
 
+def docker_rpm():
+    """
+    Returns the full NVRA of the currently-installed docker or docker-latest
+    """
+    return utils.run("rpm -q %s" % which_docker()).stdout.strip()
+
+
+def known_failures():
+    """
+    Returns a dict containing known test failures. Primary key is
+    subtest name (e.g. docker_cli/foo/bar), value is another dict
+    whose key is docker NVRA (e.g. docker-1.12.5-8.el7.x86_64),
+    value of that is a string description of the problem (e.g.
+    a bz number and comment).
+    """
+    known = {}
+    import csv
+    path_known = os.path.join(config.CONFIGCUSTOMS, 'known_failures.csv')
+    with open(path_known, 'rb') as csv_fh:
+        csv_reader = csv.reader(csv_fh)
+        for row in csv_reader:
+            if row[1] not in known:
+                known[row[1]] = {}
```

Comment:
Oooohhhh, this is my misunderstanding then, thanks for the 'diagram' form.   I had originally read 'docker NVR is the primary-key' (or mis-read).  So I thought it was ending up like:

``known['docker-1.12.5-*'] = { 'docker_cli/negativeusage/iv4': "bzetcetc" }``

okay, we're on the same page now and I agree with the structure.